app/models/api/hses.rb

```diff
@@ -1,25 +1,49 @@
+require "api_request"
 require "fake_api_response_wrapper"
 require "fake_issues"
 
-class Api::Hses
-  class Issue
-    include FakeApiResponseWrapper
-    attr_accessor :id
+module Api::Hses
+end
+
+class Api::Hses::Issue < ApiRequest
+  include FakeApiResponseWrapper
+  attr_accessor :id
+
+  def initialize(id:)
+    @id = id
+  end
+
+  # TODO update this with real data once we have /issue endpoint
+  def request
+    details_wrapper.merge(
+      data: FakeIssues.instance.json[:data].sample
+    )
+  end
+
+  private
+
+  def host
+    Rails.configuration.x.hses.api_base
+  end
+end
+
+class Api::Hses::Issues < ApiRequest
+  def initialize(user:)
+    @username = user["uid"]
+  end
+
+  def request
+    res = query("username=#{@username}&types=1")
```

Comment:
[suggestion] hmmm i don't like that we're passing through something to this method.

it would make more sense to me i think for the current `query` method to be renamed to something like 
`response` and then in this class have

```ruby
def query
  "username=#{@username}&types=1"
end
```

Reply:
Yeah that's probably cleaner, and then we can revisit once we start getting things actually passed through from parameters the user is selecting.